gazebosim/gz-rendering
# Fix transparency over heightmap

# 🦟 Bug fix

Fixes https://github.com/gazebosim/gz-rendering/issues/789

## Summary

Semi-transparent objects were always rendered behind the heightmap as seen in #789. This PR fixes the issue by setting a new render queue for objects with transparency as recommended in https://forums.ogre3d.org/viewtopic.php?t=96950

To test with gazebo:

1. launch gazebo (fortress) with the heightmap world:

```sh
ign gazebo -v 4 heightmap.sdf
```

1. Select Translate tool and click on any of the models. You should see the RGB axes rendered properly over the heightmap.

1. Right click on the model -> View > Transparent. The model should also be rendered properly over heightmap

![transparency_over_heightmap](https://user-images.githubusercontent.com/4000684/214931595-f7351864-92c5-44c0-990b-3b6797d5c2ff.png)

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.
